docs/tutorials/ros2.md

```diff
@@ -0,0 +1,184 @@
+```shell
+pixi add colcon-common-extensions
+```
+Add the created initialization script for the ros workspace to your manifest file.
+
+Then run the build command
+```shell
+pixi run colcon build
+```
+
+This will create a sourceable script in the `install` folder, you can source this script through an activation script to use your custom node.
+Normally this would be the script you add to your `.bashrc` but now you tell pixi to use it.
+
+=== "Linux & macOS"
+    ```toml title="pixi.toml"
+    [activation]
+    scripts = ["install/setup.sh"]
+    ```
+=== "Windows"
+    ```toml title="pixi.toml"
+    [activation]
+    scripts = ["install/setup.bat"]
+    ```
+??? tip "Multi platform support"
+    You can add multiple activation scripts for different platforms, so you can support multiple platforms with one project.
+    Use the following example to add support for both Linux and Windows, using the [target](../advanced/multi_platform_configuration.md#activation) syntax.
+    ```toml
+    [project]
+    platforms = ["linux-64", "win-64"]
+
+    [activation]
+    scripts = ["install/setup.sh"]
+    [target.win-64.activation]
+    scripts = ["install/setup.bat"]
+    ```
+
+Now you can run your custom node with the following command
+```shell
+pixi run ros2 run my_package my_node
+```
+
+## Simplify the user experience
+In `pixi` we have a feature called `tasks`, this allows you to define a task in your manifest file and run it with a simple command.
+Let's add a task to run the `turtlesim` example and the custom node.
+```shell
+pixi task add sim "ros2 run turtlesim turtlesim_node"
+pixi task add build "colcon build --symlink-install"
+pixi task add hello "ros2 run my_package my_node"
+```
+Now you can run these task by simply running
+```shell
+pixi run sim
+pixi run build
+pixi run hello
+```
+
+???+ tip "Advanced task usage"
+    Tasks are a powerful feature in pixi.
+
+    - You can add [`depends_on`](../advanced/advanced_tasks.md#depends-on) to the tasks to create a task chain.
+    - You can add [`cwd`](../advanced/advanced_tasks.md#working-directory) to the tasks to run the task in a different directory from the root of the project.
+    - You can add [`inputs` and `outputs`](../advanced/advanced_tasks.md#caching) to the tasks to create a task that only runs when the inputs are changed.
+    - You can use the [`target`](../configuration.md#the-target-table) syntax to run specific tasks on specific machines.
+
+    ```toml
+    [tasks]
+    sim = "ros2 run turtlesim turtlesim_node"
+    build = {cmd = "colcon build --symlink-install", inputs = ["src"]}
+    hello = { cmd = "ros2 run my_package my_node", depends_on = ["build"] }
+    ```
+
+## Build a C++ node
+To build a C++ node you need to add the `ament_cmake` and some other build dependencies to your manifest file.
+```shell
+pixi add ros-humble-ament-cmake-auto compilers pkg-config cmake
+pixi add gcc_linux-64 gxx_linux-64 make -p linux-64
```

Comment:
Why do you need to install manually `gcc_linux-64` and `gxx_linux-64` , shouldn't those be installed by `compilers` ?

Reply:
Yeah that's a mistake! Let me fix that. Still need to test on more platforms.

Reply:
Good point. Probably you also need to add `make` on all non-windows platform. To be honest, I typically install both `make` and `ninja` also on Windows just to avoid confusion. I think it would be useful to eventually add a `build-essential` or `build-essential-cpp` meta-package to conda-forge that depends on compilers and all these tools (perhaps even meson), but I never put the effort to start a discussion on this.

Reply:
Good idea, that would help to get started with conda.